pkg/util/test/util.go

```diff
@@ -2951,6 +2955,177 @@ func ValidatePrometheus(pxImageList map[string]string, cluster *corev1.StorageCl
 	return nil
 }
 
+func ValidateGrafana(pxImageList map[string]string, cluster *corev1.StorageCluster) error {
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return err
+	}
+	if opVersion.LessThan(opVer23_8) {
+		logrus.Infof("Skipping grafana validation for operation version: [%s]", opVersion.String())
+		return nil
+	}
+
+	shouldBeInstalled := cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Grafana != nil &&
+		cluster.Spec.Monitoring.Grafana.Enabled && cluster.Spec.Monitoring.Prometheus != nil &&
+		cluster.Spec.Monitoring.Prometheus.Enabled
+	err = ValidateGrafanaDeployment(cluster, shouldBeInstalled, pxImageList)
+	if err != nil {
+		return err
+	}
+	err = ValidateGrafanaService(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+	err = ValidateGrafanaConfigmaps(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaDeployment(cluster *corev1.StorageCluster, shouldBeInstalled bool, pxImageList map[string]string) error {
+	check := func() (interface{}, bool, error) {
+		pods, err := coreops.Instance().GetPods("kube-system", map[string]string{"app": "grafana"})
```

Comment:
Don't hardcode `kube-system`, this can be installed into any namespace, let's get namespace from `cluster.Namespace` and use that instead

Reply:
done